run_unittests.py

```diff
@@ -1020,7 +1020,7 @@ def test_testsetups(self):
         self._run(self.mtest_command + ['--setup=timeout'])
 
     def test_testsetup_selection(self):
-        return
+        raise unittest.SkipTest('Broken, makes the CI hang')
```

Comment:
If this is the actual problem, should it not then skip the test only if Travis or Appveyor envvars are found?

Reply:
Oh darn I thought the same thing but forgot to do it.

I was actually surprised that the original change got in at all. How did it get through review?

Reply:
Hah, it wasn't supposed to be merged, but got merged accidentally: https://github.com/mesonbuild/meson/pull/3240/files#r181593745